pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -212,7 +219,7 @@ def restore_loops(self) -> None:
                 " consider using an end of epoch checkpoint."
             )
 
-        state_dict = self._loaded_checkpoint.get("loops")
+        state_dict = self._loaded_checkpoint.pop("loops", None)
```

Comment:
I'd rather not pop here. You don't know what will the loaded checkpoint be used for later.

Reply:
Reverted !